### PR TITLE
Fix a few links in the docs

### PR DESCRIPTION
**Pull request by ollie27**
r? @steveklabnik

### CODE REVIEW

File: src/libcore/marker.rs [hunk 1]
```diff
@@ -241,8 +241,8 @@ pub trait Unsize<T: ?Sized> {
 /// compile-time error. Specifically, with structs you'll get [E0204] and with enums you'll get
 /// [E0205].
 ///
-/// [E0204]: https://doc.rust-lang.org/error-index.html#E0204
-/// [E0205]: https://doc.rust-lang.org/error-index.html#E0205
+/// [E0204]: ../../error-index.html#E0204
```
Comment (GuillaumeGomez) [inline src/libcore/marker.rs]: I don't think this is a good idea. If you build docs locally, you'' have troubles.

File: src/libcore/ops.rs [hunk 2]
```diff
@@ -182,11 +182,11 @@ pub trait Drop {
     /// After this function is over, the memory of `self` will be deallocated.
     ///
     /// This function cannot be called explicitly. This is compiler error
-    /// [0040]. However, the [`std::mem::drop`] function in the prelude can be
+    /// [E0040]. However, the [`std::mem::drop`] function in the prelude can be
     /// used to call the argument's `Drop` implementation.
     ///
-    /// [0040]: https://doc.rust-lang.org/error-index.html#E0040
-    /// [`std::mem::drop`]: https://doc.rust-lang.org/std/mem/fn.drop.html
+    /// [E0040]: ../../error-index.html#E0040
```
Comment (GuillaumeGomez) [inline src/libcore/ops.rs]: Same comment.

**Review by GuillaumeGomez — CHANGES_REQUESTED** [review]

File: src/libstd/path.rs [hunk 3]
```diff
@@ -914,6 +914,7 @@ impl<'a> cmp::Ord for Components<'a> {
 /// [`Path`]: struct.Path.html
 /// [`push`]: struct.PathBuf.html#method.push
 /// [`set_extension`]: struct.PathBuf.html#method.set_extension
+/// [`Deref`]: ../ops/trait.Deref.html
```
Comment (GuillaumeGomez) [inline src/libstd/path.rs]: Nice catch!

**Comment by GuillaumeGomez** [comment]
Thanks for your PR. Just a few things to fix and it should be good!

File: src/libcore/marker.rs (hunk 1 above)
Comment (ollie27) [inline src/libcore/marker.rs]: What do you mean? The error index is part of the local docs so this is no different to linking to the book like this.

Comment (GuillaumeGomez) [inline src/libcore/marker.rs]: If you run `make docs` (or any equivalent) then yes, but what if you just build libstd docs?

Comment (GuillaumeGomez) [inline src/libcore/marker.rs]: Hum, I see your point. I guess it's okay then.

**Review by GuillaumeGomez — APPROVED** [review]

**Pull request merged**